diff --git a/lib/desktop.template b/lib/desktop.template
index f8cf6d962..e59dc7007 100644
--- a/lib/desktop.template
+++ b/lib/desktop.template
@@ -3,8 +3,8 @@
 Name=Processing
 GenericName=Processing
 Comment=Open-source software prototyping platform
-Exec=<BINARY_LOCATION> %u
-Icon=<ICON_NAME>
+Exec=processing %u
+Icon=org.processing.processingide
 Terminal=false
 Categories=Development;IDE;Programming;
 MimeType=text/x-processing;x-scheme-handler/pde;
